avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeCache.java

```diff
@@ -308,9 +310,20 @@ private String getSchemaKey(Schema writerSchema, Schema readerSchema) {
     FastSpecificDeserializerGenerator<?> generator =
         new FastSpecificDeserializerGenerator<>(writerSchema, readerSchema, classesDir, classLoader,
             compileClassPath.orElseGet(() -> null));
-    LOGGER.info("Generated class dir: {}, and generation of specific FastDeserializer is done for writer schema: "
-            + "[\n{}\n] and reader schema: [\n{}\n]", classesDir, writerSchema.toString(true), readerSchema.toString(true));
-    return generator.generateDeserializer();
+    FastDeserializer<?> fastDeserializer = generator.generateDeserializer();
+
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Generated classes dir: {} and generation of specific FastDeserializer is done for writer schema of type: {} with fingerprint: {}"
+              + " and content: [\n{}\n] and reader schema of type: {} with fingerprint: {} and content: [\n{}\n]", classesDir, writerSchema.getFullName(),
+              writerSchema.toString(true), getSchemaFingerprint(writerSchema), readerSchema.getFullName(), getSchemaFingerprint(readerSchema),
+              readerSchema.toString(true));
+    } else {
+      LOGGER.info("Generated classes dir: {} and generation of specific FastDeserializer is done for writer schema of type: {} with fingerprint: {}"
+              + " and reader schema of type: {} with fingerprint: {}", classesDir, writerSchema.getFullName(), getSchemaFingerprint(writerSchema),
```

Comment:
getSchemaFingerprint() invokes Schema.hashCode() which (under avro 1.4) is not a cached value and hence slow. i think either ifInfoEnabled() or a decorator object who's toString() would evaluate to getSchemaFingerprint(schema) is warranted here

Reply:
In Avro-1.4,  `getFullName()` is not supported if the Schema is not a NamedSchema.
I would suggest to print the complete schema here since it will be only printed for one time...

Reply:
Efficiency shouldn't be a  problem here since this log should only print once per schema pair.
Also I am wondering how valuable to print `fingerprint` here since we could tell a schema from a fingerprint.. `FullName` might be good enough..
If `FullName` is not supported, the `type` will be printed.

Reply:
Hi

As I have mentioned before, full schema printing is somewhat troublesome from our perspective. We use a lot of massive schemas operating in different versions what results in many logging messages spanning across multiple screens of text every time the successful compilation takes place. Aesthetically speaking I think that this is a little bit too verbose on INFO level. But feedback from the (de)serializers generation process is generally appreciated. I wanted to differentiate the logging messages for the particular versions of the same schema, so that is the reason why I used the fingerprint mechanism, however I wasn't aware of the potential performance issues in avro 1.4 which can be introduced this way. But as @gaojieliu observed, does this really pose a performance threat? In vast majority of cases this should be called only once at the beginning.

Reply:
i withdraw my performance concerns here